assets/src/data/model/event/index.js

```diff
@@ -0,0 +1,115 @@
+import PropTypes from 'prop-types';
+
+/**
+ * Internal imports
+ */
+import { getQueryString as baseGetQueryString } from '../base';
+
+export const nowDateAndTime = moment();
+
+/**
+ * Described attributes for this model
+ * @type {{attributes: *}}
+ */
+export const queryDataTypes = {
+	queryData: PropTypes.shape( {
+		limit: PropTypes.number,
+		orderBy: PropTypes.oneOf( [
+			'EVT_name',
+			'EVT_ID',
+			'start_date',
+			'end_date',
+			'ticket_start',
+			'ticket_end',
+		] ),
+		order: PropTypes.oneOf( [ 'asc', 'desc' ] ),
+		showExpired: PropTypes.bool,
+		categorySlug: PropTypes.string,
+		month: PropTypes.month,
+	} ),
+};
+
+/**
+ * Default attributes for this model
+ * @type {{attributes: {limit: number, orderBy: string, order: string,
+ *   showExpired: boolean}}}
+ */
+export const defaultQueryData = {
+	queryData: {
+		limit: 100,
+		orderBy: 'start_date',
+		order: 'desc',
+		showExpired: false,
+	},
+};
+
+/**
+ * Used to map an orderBy string to the actual value used in a REST query from
+ * the context of an event.
+ *
+ * @param {string} orderBy
+ *
+ * @return { string } Returns an actual orderBy string for the REST query for
+ *                      the provided alias
+ */
+export const mapOrderBy = ( orderBy ) => {
+	const orderByMap = {
+		start_date: 'Datetime.DTT_EVT_start',
+		end_date: 'Datetime.DTT_EVT_end',
+		ticket_start: 'Datetime.Ticket.TKT_start_date',
+		ticket_end: 'Datetime.Ticket.TKT_end_date',
+	};
+	return isUndefined( orderByMap[ orderBy ] ) ?
+		orderBy :
+		orderByMap[ orderBy ];
+};
+
+/**
+ * Builds where conditions for an events endpoint request using provided
+ * information.
+ *
+ * @param {boolean} showExpired  Whether or not to include expired events.
+ * @param {string} categorySlug  Return events for the given categorySlug
+ * @param {string} month         Return events for the given month.  Can be any
+ *                                 in any month format recognized by moment.
+ * @return {string}             The assembled where conditions.
+ */
+export const whereConditions = ( { showExpired = true, categorySlug, month = 'none' } ) => {
+	const where = [];
+	const GREATER_AND_EQUAL = encodeURIComponent( '>=' );
+	const LESS_AND_EQUAL = encodeURIComponent( '<=' );
+
+	if ( ! showExpired ) {
+		where.push( 'where[Datetime.DTT_EVT_end**expired][]=>&where[Datetime.DTT_EVT_end**expired][]=' +
+			nowDateAndTime.local().format() );
+	}
+	if ( categorySlug ) {
+		where.push( 'where[Term_Relationship.Term_Taxonomy.Term.slug]=' + categorySlug );
+	}
+	if ( month && month !== 'none' ) {
+		where.push( 'where[Datetime.DTT_EVT_start][]=' +
+			GREATER_AND_EQUAL +
+			'&where[Datetime.DTT_EVT_start][]=' +
+			moment().month( month ).startOf( 'month' ).local().format() );
+		where.push( 'where[Datetime.DTT_EVT_end][]=' +
+			LESS_AND_EQUAL +
+			'&where[Datetime.DTT_EVT_end][]=' +
+			moment().month( month ).endOf( 'month' ).local().format() );
+	}
+	return where.join( '&' );
```

Comment:
If a component or block requires a where condition that is not here, then in your opinion should we:

- add the new condition and corresponding prop to this method and everywhere else it is required
- override this method (or another) and simply append the additional where condition to the query string

Reply:
One of the reasons I extracted these into their own files is so that they could be considered the "authority" for parsing where conditions for that model.  So for now, the option would be "add the new condition and corresponding prop to this method and everywhere else it is required".  

However, as we see patterns emerge, we may find that we can autogenerate some of these a bit (or at least start with a common base).